build/plugins/attributes.html

```diff
@@ -0,0 +1,132 @@
+<!-- attributes plugin -->
+
+<script>
+    (function() {
+        // /////////////////////////
+        // DESCRIPTION
+        // /////////////////////////
+
+        // This Manubot plugin allows arbitrary HTML attributes to be attached
+        // to (almost) any element. Place an HTML comment inside or next to the
+        // desired element in the format <!-- attribute="value" -->
+
+        // /////////////////////////
+        // OPTIONS
+        // /////////////////////////
+
+        // plugin name prefix for url parameters
+        const pluginName = 'attributes';
+
+        // default plugin options
+        const options = {
+            // whether plugin is on or not
+            enabled: 'true'
+        };
+
+        // change options above, or override with url parameter, eg:
+        // 'manuscript.html?pluginName-enabled=false'
+
+        // /////////////////////////
+        // SCRIPT
+        // /////////////////////////
+
+        // start script
+        function start() {
+            // get list of comments in document
+            const comments = findComments();
+
+            for(const comment of comments) {
+                // if comment doesn't have valid parent element
+                if (!comment.parentElement)
+                    continue;
+
+                // get text content of comment
+                let text = comment.nodeValue.trim();
+
+                // regex's for finding attribute/value pairs in the format of
+                // attribute="value" or attribute='value
+                const regex2 = /([a-zA-Z\-]+)?=\"(.+?)\"/;
+                const regex1 = /([a-zA-Z\-]+)?=\'(.+?)\'/;
+
+                // loop through attribute/value pairs
+                let match;
+                let count = 0;
+                while(
+                    match = text.match(regex2) || text.match(regex1),
+                    count++,
+                    match && count < 100
```

Comment:
What is the significance of 100 here?

Reply:
This is an arbitrary safety limit so we don't crash the tab if anything accidentally goes wrong in the while loop. I don't know all the possible edge cases of what the javascript `match` function can return, so it's slightly possible that the "remove from string" step might fail in a way that causes an infinite loop.

However, there's while loops in the other plugins, and I don't do this sort of safety limit. Those I feel a little more confident about (that they're predictable), but perhaps I should consider putting safety cutoffs in those cases as well.